src/composables/useFlawModel.ts

```diff
@@ -62,15 +65,28 @@ export function useFlawModel(forFlaw: ZodFlawType = blankFlaw(), onSaveSuccess:
 
     await postFlaw(flawForPost)
       .then(createSuccessHandler({ title: 'Success!', body: 'Flaw created' }))
-      .then((response: any) => {
+      .then(async (response: any) => {
+        if (response.uuid) {
+          flaw.value.uuid = response.uuid;
+          saveDraftFlaw(flaw.value);
+          try {
+            if (flaw.value.acknowledgments.length > 0) {
+              await flawAttributionsModel.saveAcknowledgments(flaw.value.acknowledgments);
+            }
+            if (flaw.value.references.length > 0) {
+              await flawAttributionsModel.saveReferences(flaw.value.references);
+            }
+          } catch (error: any) {
+            createCatchHandler('Error creating Flaw attributions');
+          }
+        }
         router.push({
           name: 'flaw-details',
           params: { id: response?.cve_id || response?.uuid },
         });
-        flaw.value.uuid = response.uuid;
       })
-      .catch(createCatchHandler('Error creating Flaw'));
-
+      .catch(createCatchHandler('Error creating Flaw'))
+      .finally(() => isSaving.value = false);
     if (wasCvssModified.value) {
       await saveCvssScores();
     }
```

Comment:
I would consider only saving acknowledgments/references if any have been added, and putting it in a separate step from the initial flaw save

Reply:
They are being saved only if any was added. That's handled by lines 73 & 76:
` if (flaw.value.acknowledgments.length > 0) ` ... `if (flaw.value.references.length > 0)` 

As it is flaw creation it is not needed to consider existent attributions. If there is any value will be a new one that needs to be saved.

Is that what you were referring to?

Reply:
Here is what I was thinking...

```js
    const response = await postFlaw(flawForPost)
      .then(createSuccessHandler({ title: 'Success!', body: 'Flaw created' }))
      .then(async (response: any) => {
        router.push({
          name: 'flaw-details',
          params: { id: response?.cve_id || response?.uuid },
        });
        // We could also update the flaw with the response here, this would impact the refresh logic so as not to rely on the router for refreshing the flaw, which could be a better approach
      })
      .catch(createCatchHandler('Error creating Flaw'))
      .finally(() => isSaving.value = false);

    // This is the separate step I was referring to earlier
    if (response.uuid) {
      flaw.value.uuid = response.uuid;
      saveDraftFlaw(flaw.value);

      if (flaw.value.acknowledgments.length > 0) {
        await flawAttributionsModel.saveAcknowledgments(flaw.value.acknowledgments)
        .catch(createCatchHandler('Error creating Flaw acknowledgments'));
      }
      if (flaw.value.references.length > 0) {
        await flawAttributionsModel.saveReferences(flaw.value.references)
        .catch(createCatchHandler('Error creating Flaw references'));
      }
    }

    if (wasCvssModified.value) {
      await saveCvssScores();
    }
    isSaving.value = false;
```

Reply:
I get your point on separating the logic but I don't see what practical benefits we get from that.
IMO refreshing the flaw from the response will complicate more the logic, than just doing the redirect from `flaw-create` to `flaw-details` using the router.